sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JacksonParser.scala

```diff
@@ -100,34 +101,27 @@ object JacksonParser {
         parser.getFloatValue
 
       case (VALUE_STRING, FloatType) =>
-        // Special case handling for NaN and Infinity.
```

Comment:
yea, should revert it back. BTW,  do we actually test "inf" and "-inf" before? Because "inf".toFloat is not legal.